### PR TITLE
Fix 02116_tuple_element with Analyzer

### LINKED ISSUE #51667: [Analyzer] flaky test: 02116_tuple_element
https://s3.amazonaws.com/clickhouse-test-reports/0/3b7f6a43243913f302f8c19c664e07ac6ee790e4/stateless_tests__release__analyzer_.html

```
2023-06-30 04:58:10 Code: 10. DB::Exception: Received from localhost:9000. DB::Exception: Tuple doesn't have element with index '0': In scope SELECT t1.0 FROM t_tuple_element. (NOT_FOUND_COLUMN_IN_BLOCK)
2023-06-30 04:58:10 (query: SELECT tupleElement(t1, 0) FROM t_tuple_element; -- { serverError ILLEGAL_INDEX })
```

### PR DESCRIPTION
**Pull request by rschu1ze**
Fixes #51667

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
This is an automated comment for commit 65b82dea47ae98da57bbd788f8844c9d5d17c987 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51669/65b82dea47ae98da57bbd788f8844c9d5d17c987/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by qoega — APPROVED** [review]

**Comment by qoega** [comment]
Closes #51667

File: tests/queries/0_stateless/02116_tuple_element.sql [hunk 1]
```diff
@@ -17,8 +17,10 @@ EXPLAIN SYNTAX SELECT tupleElement(t1, 'a') FROM t_tuple_element;
 SELECT tupleElement(number, 1) FROM numbers(1); -- { serverError ILLEGAL_TYPE_OF_ARGUMENT }
 SELECT tupleElement(t1) FROM t_tuple_element; -- { serverError NUMBER_OF_ARGUMENTS_DOESNT_MATCH }
 SELECT tupleElement(t1, 'b') FROM t_tuple_element; -- { serverError NOT_FOUND_COLUMN_IN_BLOCK, UNKNOWN_IDENTIFIER }
-SELECT tupleElement(t1, 0) FROM t_tuple_element; -- { serverError ILLEGAL_INDEX }
-SELECT tupleElement(t1, 3) FROM t_tuple_element; -- { serverError ILLEGAL_INDEX }
+SELECT tupleElement(t1, 0) FROM t_tuple_element SETTINGS allow_experimental_analyzer = 0; -- { serverError ILLEGAL_INDEX }
```
Comment (Algunenano) [inline tests/queries/0_stateless/02116_tuple_element.sql:20]: Note that you can pass several errors in the comment to be accepted, e.g. `-- { serverError ILLEGAL_INDEX, NOT_FOUND_COLUMN_IN_BLOCK }`. That way you don't need to manually test with and without the analyzer (and leave that to global config)

Comment (rschu1ze) [reply tests/queries/0_stateless/02116_tuple_element.sql]: Good idea. I actually saw this syntax yesterday in some other test and wondered if the author made a mistake :-)

**Review by alexey-milovidov — APPROVED** [review]

**Comment by tavplubix** [comment]
Integration tests were broken due to docker images update

**Pull request merged**